xarray/core/dataarray.py

```diff
@@ -1736,5 +1736,93 @@ def dot(self, other):
 
         return type(self)(new_data, new_coords, new_dims)
 
+    def quantile(self, q, dim=None, axis=None, interpolation='linear'):
+        """
+
+        Compute the qth quantile of the data along the specified axis,
+        while ignoring nan values.
+        Returns the qth quantiles(s) of the array elements.
+
+        Parameters
+        ----------
+        q : float in range of [0,100] (or sequence of floats)
```

Comment:
Pandas uses quantiles between 0 and 1. Since we're using the name `quantile` (rather than `percentile`), I would stick with that convention (just divide by 100 to use `nanpercentile` under the hood).